docs/source/contributing/index.rst

```diff
@@ -1,5 +1,5 @@
-Contributing to CSET
-====================
+Contributing to CSET as a developer
+===================================
```

Comment:
```suggestion
Developer's Guide
=================
```
How does this sound? It is what iris calls the section in their docs.
![image](https://github.com/MetOffice/CSET/assets/113985160/68fdb21d-82b9-4591-ada8-830f84536fa7)